src/ssa/malloc_ssa.cpp

```diff
@@ -353,6 +353,48 @@ bool replace_malloc(
   return result;
 }
 
+/// Replaces the RHS of the following instruction if it is an assignemnt
+/// and its LHS is equal to name. Returns whether something was changed.
+bool set_following_assign_to_true(
+  std::list<goto_programt::instructiont>::iterator it,
+  std::list<goto_programt::instructiont>::iterator end,
+  const std::string &name)
+{
+  bool result=false;
+  for(; it!=end; it++)
```

Comment:
Not sure if it is the best idea to iterate up to the end of the GOTO program. There may be branching and we may fail to set the variable in all reachable branches.

I'd suggest checking only the following location or iterating up to the first goto/dead.

Reply:
The loop is stopped on matching `dead` instruction. Lets also add break on `goto` that should be safer and will cover the  problem it aimed to solve